build-mac/CMakeLists.txt

```diff
@@ -101,9 +101,33 @@ set(STARROCKS_THIRDPARTY "$ENV{STARROCKS_THIRDPARTY}")
 set(SRC_DIR "${BASE_DIR}/src")
 set(GENSRC_DIR "${SRC_DIR}/gen_cpp/build")
 set(GENCODE_DIR "${GENSRC_DIR}/gen_cpp")
+set(GEN_THRIFT_SENTINEL "${GENCODE_DIR}/AgentService_types.cpp")
+
+# On macOS we copy pre-generated thrift/protobuf sources into be/src/gen_cpp.
+# When gensrc/build is unavailable (common outside Linux), fall back to the
+# checked-in copies so the linker still sees the thrift symbol definitions.
+if(NOT EXISTS "${GENCODE_DIR}")
```

Comment:
https://github.com/StarRocks/starrocks/pull/66509/files

I think this pr should fix this issue, could you check again?